lib/foreman/gettext/support.rb

```diff
@@ -42,19 +42,29 @@ def self.add_text_domain(locale_domain, locale_dir)
           :report_warning => false
       end
 
-      def self.register_human_localenames
-        original_locale = FastGettext.locale
-        FastGettext.class.class_eval { attr_accessor :human_available_locales }
-        FastGettext.human_available_locales = []
-        FastGettext.default_available_locales.sort.each do |locale|
-          FastGettext.locale = locale
-          # TRANSLATORS: Provide locale name in native language (e.g. English, Deutsch or Portugues)
-          human_locale = _("locale_name")
-          human_locale = locale if human_locale == "locale_name"
-          FastGettext.human_available_locales << [human_locale, locale]
+      # Loading all available locales to get the language translation is slow
+      # (2 seconds for 20 languages), so let's lazy load that.
+      FastGettext.class.class_eval do
```

Comment:
This is bad. `FastGettext.class` is `Module` and all the methods defined in the `class_eval` block will be available on *any* module. This is almost surely not what we want.

```
[1] pry(main)> show-source Foreman.human_available_locales

From: /home/aruzicka/work/redhat/foreman/lib/foreman/gettext/support.rb:50:
Owner: Module
Visibility: public
Signature: human_available_locales()
Number of lines: 18

def human_available_locales
  original_locale = FastGettext.locale
  mutex.synchronize do
    return locales if locales
    Rails.logger.debug "Loading and caching all available locales"
    locales = []
    FastGettext.default_available_locales.sort.each do |locale|
      FastGettext.locale = locale
      # TRANSLATORS: Provide locale name in native language (e.g. Deutsch instead of German)
      human_locale = _("locale_name")
      human_locale = locale if human_locale == "locale_name"
      locales << [human_locale, locale]
    end
    locales
  end
ensure
  FastGettext.locale = original_locale
end
```

Could we instead reopen the module (`module FastGettext`) and then access its eigenclass with `class << self` instead of patching every single module in existence? In the `class << self` block we could probably still use `attr_accessor :locales` and define class instance variable with accessor for the mutex.